src/mds/MDBalancer.cc

```diff
@@ -150,6 +150,15 @@ void MDBalancer::handle_export_pins(void)
 
   set<CDir *> authsubs;
   mds->mdcache->get_auth_subtrees(authsubs);
+
+  if (authsubs.size() > AUTH_TREES_THRESHOLD) {
```

Comment:
While testing the other commit on this PR, I observed that an "auth tree" was being reported every few seconds; so wrote a similar patch for it too.

For the sake of book-keeping should I create an issue on the tracker?

Reply:
It's fine to include it in this PR/issue.

I think even 50 is too high. Maybe 5?

Reply:
Would it be nice to check the auth tree's depth too? We are doing it for subtrees but not for auth trees.

Reply:
IIUC, the subtree depth doesn't factor in here. `authsubs` contains all authoritative subtrees at all depths.